en/middleware/micrortps.md

```diff
@@ -30,58 +32,84 @@ The main elements of the architecture are the client and agent processes shown i
 - The *Client* is PX4 middleware daemon process that runs on the flight controller. It subscribes to uORB topics published by other PX4 components and sends any updates to the *Agent* (via a UART or UDP port). It also receives messages from the *Agent* and publishes them as uORB message on PX4.
 - The *Agent* runs as a daemon process on an offboard computer. It watches for uORB update messages from the *Client* and (re)publishes them over RTPS. It also subscribes to "uORB" RTPS messages from other RTPS applications and forwards them to the *Client*.
 - The *Agent* and *Client* are connected via a serial link (UART) or UDP network. The uORB information is [CDR serialized](https://en.wikipedia.org/wiki/Common_Data_Representation) for sending (*CDR serialization* provides a common format for exchanging serial data between different platforms).
-- The *Agent* and any *FastRTPS* applications are connected via UDP, and may be on the same or another device. In a typical configuration they will both be on the same system (e.g. a development computer, Linux companion computer or compute board), connected to the *Client* over a Wifi link or via USB.
+- The *Agent* and any *Fast-RTPS* applications are connected via UDP, and may be on the same or another device. In a typical configuration they will both be on the same system (e.g. a development computer, Linux companion computer or compute board), connected to the *Client* over a Wifi link or via USB.
+
+
+## Architectural overview for a ROS2/ROS application pipeline
+
+<!--![basic example flow](../../assets/middleware/micrortps/architecture_ros.png)-->
+
+ROS2 has been developed on top of the DDS/RTPS, which is what composes its middleware. This same middleware serves as the end-to-end architecture for plugging different applications that rely on distributed discovery, serialization and QoS control over the transportation layer.
+
+So one is able to publish and subscribe to uORB data from PX4 using ROS2 (and ROS1 if using the [`ros1_bridge`](https://github.com/ros2/ros1_bridge), it just needs to create it's own listener our advertiser nodes in the respective frameworks. The only consideration to take into account is that the msg types, headers and source files being used on both client and agent side (and consequently, on the ROS side) need to be the same and generated from the same IDL (Interface Description Language).
```

Comment:
This paragraph is confusing
- the brackets don't match up, and without a diagram I can't see how it "should" be. 
- Typo: "it just needs to create it's own listener our advertiser nodes in the respective frameworks." - I think you mean "and advertiser nodes"
- I can't see where ros1_bridge fits in. I would pull ROS 1 out into its own paragraph. 
- We say "e only consideration to take into account". Isn't this always the same consideration? 
- `it just needs to create it's own listener our advertiser nodes in the respective frameworks.` - what is "IT" here? ROS2 needs to create its own nodes? What do you mean by "listener and advertiser nodes"? Do you mean uOrb topics?

Here is a _guess_ at what I think you want to say in this section:

```
RTPS is the ROS2 native communications middleware, which makes it easy to integrate with PX4.

In order to publish and subscribe to PX4 uORB data from ROS2 (and visa versa) you will need to create matching uORB topics in PX4 and listener/advertiser in ROS2. The msg types, headers and source files used on both client and agent side (and consequently, on the ROS side) must be generated from the same IDL (Interface Description Language) definition.

ROS1 integration with is also supported via the [ros1_bridge](https://github.com/ros2/ros1_bridge)  <NEED DIAGRAM TO SEE WHERE THAT FITS>
```

Reply:
Well actually it's the other way around: you don't mess with the the uORB topics. It's the `px4_ros_com` package that takes the available uORB topic messages and generates the respective sources and headers for the ROS2/ROS messages and also the IDL files required to generate *micrortps_agent*. I will redo this paragraph to make it more clear.